pageserver/src/waldecoder.rs

```diff
@@ -81,38 +102,46 @@ impl WalStreamDecoder {
 
     // Returns a tuple:
     // (end LSN, record)
-    pub fn poll_decode(&mut self) -> Option<(u64, Bytes)> {
+    pub fn poll_decode(&mut self) -> Result<Option<(u64, Bytes)>, WalDecodeError> {
```

Comment:
Use `///` to document functions?

This is a pretty hairy return type, can you document what it means? How is `Ok(None)` different from `Err`?

Reply:
Added a comment